### ISSUE TITLE
Make test gives an error on Windows 10 msys2

### ISSUE THREAD

**Issue by zenntenn** [idris-lang/Idris2#399]
This issue comes up in the tests for the latest HEAD running on MinGW 64-Bit, as a part of make test, with idris2/perf003

Error message:

`idris2/perf003: succesException in collect: cannot collect when multiple threads are active
rm: cannot remove 'libmkalloc.dll': Device or resource busy
make[2]: *** [Makefile:25: clean] Error 1`

However, it still says at the end `264/264 tests successful`

**Comment by zenntenn** [comment]
Note, `make test only=idris2/perf003` and `make test only=idris2/perf004` do not replicate the issue, neither does `make test only=idris2/perf`

**Comment by andorp** [comment]
Observed on a debian machine with
```
chez/chez031: success
Exception in collect: cannot collect when multiple threads are active
```
Versions
```
cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"

chezscheme9.5 --version
9.5
```

**Comment by cypheon** [comment]
I'm also hitting this issue on FreeBSD with Chez 10.2.0

It seems @mjustus already had a go at this last year, but that change never made it into `main`.

I submitted a fix that should solve this issue here, along with an explanation of the root cause: #3643

**Comment by mjustus** [comment]
> It seems [@mjustus](https://github.com/mjustus) already had a go at this last year, but that change never made it into `main`.

That's true. I never turned it inro a PR because I knew too little about the Chez's GC to have any confidence in the change. I think I left a comment on the Idris Discord soliciting help.
